hw2/02-reverse.js

```diff
@@ -1,3 +1,25 @@
 /** Exercise 02 - Reverse **/
 
-// Add your code here
+let number_input = document.getElementById("input");
+let submit_button = document.getElementById("reverse");
+let out = document.createElement('div');
+submit_button.parentElement.append(out);
+
+submit_button.addEventListener("click", () => {   //If there is no form, the "submit" action is not triggered, so "submit" as the action in event listener would not work.
+
+    out.style.margin = "1.5em 0 0.5em 0";
+
+    //Reverse the number and output it
+    const number = number_input.value.split("");            
+    if (number.length === 8)
+    {
+      let reversed = number.reverse().join("");
+      out.textContent = `${number_input.value} --> ${reversed}`;
```

Comment:
Also, for this line, if I replaced `number_input.value` with `number` (declared on line 13 and should not be reversed), the output is still reversed somehow. Does anyone know what I'm missing? I guess that line 16 could be changing `number` with `reverse()`, but I thought it would only return a reversed array to `reversed`.

Reply:
I see that your code looks fine, can you explain more about what you are concerning? Can you give an example about that?

Reply:
I figured it out. Line 16 is where I was having issues because my number would stay reversed. I didn't read the documentation correctly, but reverse() is an _in place_ function that reverses the array, so that's why number would output reversed. Oops!